tests/cache/test-vary-cache.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Automattic\VIP\Cache;
+
+class Vary_Cache_Test extends \WP_UnitTestCase {
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+
+		require_once( __DIR__ . '/../../cache/class-vary-cache.php' );
+	}
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->original_COOKIE = $_COOKIE;
+	}
+
+	public function tearDown() {
+		$_COOKIE = $this->original_COOKIE;
+		parent::tearDown();
+	}
+
+	/**
+	 * Helper function for accessing protected methods.
+	 */
+	protected static function get_method( $name ) {
+		$class = new \ReflectionClass( __NAMESPACE__ . '\API_Client' );
+		$method = $class->getMethod( $name );
+		$method->setAccessible( true );
+		return $method;
+	}
+
+	public function get_test_data__is_user_in_group_segment() {
+		return [
+			'group-not-defined' => [
+				[],
+				[],
+				'dev-group',
+				'yes',
+				false,
+			],
+
+			'user-not-in-group' => [
+				[
+					'vip-go-seg' => 'design-group_--_yes',
+				],
+				[
+					'dev-group',
+				],
+				'dev-group',
+				'yes',
+				false,
+			],
+
+			'user-in-group-with-empty-segment' => [
+				[
+					'vip-go-seg' => 'dev-group_--_',
+				],
+				[
+					'dev-group',
+				],
+				'dev-group',
+				'',
+				true,
```

Comment:
Should an empty segment mean that the user is in the group? What would be a use case for that?

Reply:
Nothing specific but I think if the user explicitly sets a group with an empty value and then checks for it, we should match it. Unless we want to make `value` mandatory